spec/models/order_detail_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+RSpec.describe OrderDetail, type: :model do
+  let!(:order) {FactoryBot.create :order}
+  let!(:product) {FactoryBot.create :product}
+  subject {FactoryBot.build(:order_detail, order_id: order.id, product_id: product.id)}
+
+  describe "Database" do
+    it {expect(subject).to have_db_column(:order_id).of_type :integer}
+    it {expect(subject).to have_db_column(:product_id).of_type :integer}
+    it {expect(subject).to have_db_column(:price).of_type :integer}
+    it {expect(subject).to have_db_column(:quantily).of_type :integer}
+    it {expect(subject).to have_db_column(:amount).of_type :integer}
+  end
+
+  describe "Create" do
+    it {expect(subject).to be_valid}
+  end
+
+  describe "Associations" do
+    it {expect(subject.order).to eq(order)}
+    it {expect(subject.product).to eq(product)}
```

Comment:
test associations nên test kiểu như này
```suggestion
    it {should belong_to :order}
```